### PR TITLE
feat(player): subtitle settings + refactor + crash fixes

### LINKED ISSUE #941: Subtitle Customization
### Describe your suggested feature

For extensions with soft subs can basic subtitle customization be added such as font size, font family, shadow, opacity, etc?

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/jmir1/aniyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.12.3.10](https://github.com/jmir1/aniyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Quickdesh**
Changes
---
- Refactored every player sheet to use compose
- Changed screenshot sheet to replicate the one used in upstream (future proofing)
- Added subtitle settings dialog (with @abdallahmehiz)
    - Delay, Font and Color pages
    - Change audio and subtitle delay
    - Change Subtitle font size, bold and italics style
    - Change text, border and background color of subtitle text
- Cleaned up mpv statistics page preference (future proofing)
- Cleaned up other player dialogs' code
- Fixed a bug with mpv files' speed and video-aspect
- Possibly have fixed 2 major bugs related to the player leading to a lot less crashing

Screenshots
---
Screenshot_20231013-212628

Screenshot_20231013-212621

Screenshot_20231013-210949

Screenshot_20231013-212603

Screenshot_20231013-212605

Screenshot_20231013-212610

Screenshot_20231013-212616

References
---

- Closes #941
- Closes #961

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/ui/player/settings/PlayerSettingsScreenModel.kt [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    @Composable
+    fun OverrideSubtitlesSwitch(
+        content: @Composable () -> Unit,
+    ) {
+        val overrideSubtitles by preferences.overrideSubtitlesStyle().collectAsState()
+
+        val updateOverride = {
+            val overrideType = if (overrideSubtitles) "no" else "force"
+            togglePreference(PlayerPreferences::overrideSubtitlesStyle)
+            MPVLib.setPropertyString("sub-ass-override", overrideType)
```
Comment (jmir1) [inline app/src/main/java/eu/kanade/tachiyomi/ui/player/settings/PlayerSettingsScreenModel.kt:86]: I think we should let the user change subtitle styles *without* forcing them to override ASS subtitles.
ASS subs can look quite bad when you override their styling, whereas typical WebVTT subtitles can be styled by the user exactly to their liking.

Comment (jmir1) [reply app/src/main/java/eu/kanade/tachiyomi/ui/player/settings/PlayerSettingsScreenModel.kt:86]: Maybe the toggle should read "Override ASS subtitle styles" and be below the sub settings, while the sub settings are always enabled and pre-filled with the default values.
Maybe there should also be a button to "Reset to default style".

**Comment by abdallahmehiz** [comment]
why didn't ya make it like this tho
Untitled-1

**Comment by jmir1** [comment]
> why didn't ya make it like this tho Untitled-1

This looks like shit in portrait mode, I think the current way is fine

**Review by jmir1 — DISMISSED** [review]

**Pull request merged**